### DISCUSSION TITLE
Cypress tests not working in CI with Angular

### DISCUSSION THREAD

**Discussion by rasfa98** [DevSkillsHQ/discusssions#11] [Help] [answered]
Hello! 👋 

When I imported my boilerplate `fullstack-boilerplate-dotnet-angular` I was using Node v.19. This caused an issue where I could not run the tests locally always getting `Error: connect ECONNREFUSED 127.0.0.1:4200`. I then cleared my `node_modules` folder and removed my `package-lock.json` and installed the dependencies using Node v16 and it turned out to work as expected.

I am now facing the problem where I am getting the same issue in CI even if the `package-lock.json` file has been pushed. I do not know if there is an issue on my end with some version or if the tests run before the Angular server is started. If anyone has experienced the same issue or have any idea please let me know 🙂 

![Skärmavbild 2023-03-19 kl  14 13 01](https://user-images.githubusercontent.com/31412046/226178085-75f102dc-f55c-41e1-8baa-e625b97c8980.png)

**Comment by fenske** [comment]
Hi @rasfa98. I'm Anton from DevSkills. Please don't spend more time on this. 

We'll look into it and post the results here once we've investigated.

Will make a corresponding note to the hiring team about this issue so don't worry about it.

I wish you the best of the luck with the rest of your interview process!

**Reply by rasfa98** [reply]
Thanks! 🙂

**Reply by fenske** [reply]
Referenced the answer below. Don't forget to create a PR. 🙂

All the best!

**Reply by rasfa98** [reply]
Works like a charm! 👍  Thanks for the quick support 🙂

**Comment by fenske** [comment, marked as answer]
Just pushed an update that fixed the issue which was about running the Angular app with `ng serve --host 0.0.0.0 --port 4200` (it instructed the Angular dev server to bind to all available network interfaces instead of the default localhost (which is equivalent to 127.0.0.1)).

Some background. When binding to localhost or 127.0.0.1, the server is only accessible from the same machine it's running on. In your local development environment, this is usually sufficient, as both your application and the Cypress tests are running on the same machine.

However, when running the tests in GitHub Actions, the Angular server and the Cypress tests might be running in separate containers or networks. In such cases, binding the server to localhost can prevent the Cypress tests from accessing the server, causing connection issues.

By changing the host to 0.0.0.0, it allowed the Angular server to bind to all available network interfaces, making it accessible from any IP address, including the one used by the Cypress tests in the CI environment.
